src/compile/mark/area.ts

```diff
@@ -19,65 +19,134 @@ export namespace area {
 
     const stack = model.stack();
     const xFieldDef = model.encoding().x;
+    const x2FieldDef = model.encoding().x2;
     // x
     if (stack && X === stack.fieldChannel) { // Stacked Measure
       p.x = {
         scale: model.scaleName(X),
         field: model.field(X, { suffix: '_start' })
       };
-    } else if (isMeasure(xFieldDef)) { // Measure
-      p.x = { scale: model.scaleName(X), field: model.field(X) };
-    } else if (isDimension(xFieldDef)) {
-      p.x = {
-        scale: model.scaleName(X),
-        field: model.field(X, { binSuffix: '_mid' })
-      };
-    }
-
-    // x2
-    if (orient === 'horizontal') {
-      if (stack && X === stack.fieldChannel) {
+      if (orient === 'horizontal') {
         p.x2 = {
           scale: model.scaleName(X),
           field: model.field(X, { suffix: '_end' })
         };
+      }
+    } else if (isMeasure(xFieldDef) || isMeasure(x2FieldDef)) { // Measure
```

Comment:
why do you need to group `x` and `x2` logic together?  

The old code decouples `x` and `x2`  (same for `y` / `y2`).